### PR TITLE
LLM: refactor transformers and langchain class name

### PR DESCRIPTION
**Pull request by plusbang**
## Description

Related to https://github.com/analytics-zoo/nano/issues/410

### 2. User API changes

- native int4:

previuos:
```python
from bigdl.llm.transformers import BigdlForCausalLM
```
**New**:
```python
from bigdl.llm.transformers import BigdlNativeForCausalLM
```

- langchain with native int4:
Previous:
```python
from bigdl.llm.langchain.llms import BigdlLLM
from bigdl.llm.langchain.embeddings import BigdlLLMEmbeddings
```
**New**:
```python
from bigdl.llm.langchain.llms import BigdlNativeLLM
from bigdl.llm.langchain.embeddings import BigdlNativeEmbeddings
```

### 3. Summary of the change 

- change class names (native int4, langchain with native int4) in source code
- update examples
- update uts

### 4. How to test?
- [x] Unit test

### CODE REVIEW

File: python/llm/README.md [hunk 1]
```diff
@@ -115,8 +115,8 @@ You may run the models using `transformers`-style API in `bigdl-llm`.
       outfile='/path/to/output/', outtype='int4', model_family="llama")
```
Comment (jason-dai) [inline python/llm/README.md:115]: BTW, the indentation of "- ##### Using native INT4 format" above is wrong

**Comment by jason-dai** [comment]
LGTM

File: python/llm/README.md (hunk 1 above)
Comment (plusbang) [reply python/llm/README.md:115]: > BTW, the indentation of "- ##### Using native INT4 format" above is wrong

oh, I modified the indentation now, And another typo `lm-convert` in readme is also fixed.

**Review by rnwang04 — APPROVED** [review]
LGTM

**Pull request merged**